rcl/src/rcl/remap_impl.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__REMAP_IMPL_H_
+#define RCL__REMAP_IMPL_H_
+
+#include "rcl/types.h"
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+/// enum doubles as a bitmask for rule sthat apply to both topics and services
+typedef enum rcl_remap_type_t
+{
+  RCL_UNKNOWN_REMAP = 0,
+  RCL_TOPIC_REMAP = 1u << 0,
+  RCL_SERVICE_REMAP = 1u << 1,
+  RCL_NODENAME_REMAP = 1u << 2,
+  RCL_NAMESPACE_REMAP = 1u << 3
+} rcl_remap_type_t;
+
+typedef struct rcl_remap_t
+{
+  /// bitmask indicating what type of rule this is
+  rcl_remap_type_t type;
+  /// a node name that this rule is limited to, or NULL if it applies to any node
+  char * node_name;
+  /// match portion of a rule, or NULL if node name or namespace replacement
+  char * match;
+  /// replacement portion of a rule
+  char * replacement;
+} rcl_remap_t;
+
+/// Get an rcl_remap_t structure initialized with NULL
+rcl_remap_t
+rcl_remap_get_zero_initialized();
+
+/// Deallocate memory used in an rcl_remap_t structure
```

Comment:
Same comment about Deallocate being slightly misleading.

Reply:
"reclaim" in 6ac891fd7a79a70e9f87926aefccbc48b379c7a3